webui/src/pages/repositories/repository/changes.jsx

```diff
@@ -198,57 +201,121 @@ const ChangesBrowser = ({ repo, reference, after, onSelectRef, onPaginate }) =>
+                                        showActions={true}
+                                        onRevert={(entry) => {
+                                            branches
+                                                .revert(repo.id, reference.id, {type: entry.path_type, path: entry.path})
+                                                .then(() => {
+                                                    setInternalRefresh(!internalRefresh)
+                                                })
+                                                .catch(error => {
+                                                    setActionError(error)
+                                                })
+                                        }}
+                                        onNavigate={entry => {
+                                            return {
+                                                pathname: '/repositories/:repoId/changes',
+                                                params: {repoId: repo.id},
+                                                query: {
+                                                    delimiter: "/",
+                                                    prefix: entry.path,
+                                                    ref: reference.id
+                                                }
+                                            }
+                                        }}
+                                    />
                                 ))}
                                 </tbody>
                             </Table>
-                        </Card>
-                    )}
 
-                <Paginator onPaginate={onPaginate} nextPage={nextPage} after={after}/>
+
+                            <Paginator onPaginate={onPaginate} nextPage={nextPage} after={after}/>
+
+                    </Card.Body>
+                </Card>
+                )}
             </div>
+
         </>
     )
 }
 
 const ChangesContainer = () => {
     const router = useRouter();
     const { repo, reference, loading, error } = useRefs()
-    const { after } = router.query
+    const { after, prefix, delimiter } = router.query
 
     if (loading) return <Loading/>
     if (!!error) return <Error error={error}/>
 
     return (
         <ChangesBrowser
             after={(!!after) ? after : ""}
+            prefix={(!!prefix) ? prefix : ""}
+            delimiter={(!!delimiter) ? delimiter : ""}
             repo={repo}
             reference={reference}
             onPaginate={after => router.push({
                 pathname: `/repositories/:repoId/changes`,
                 params: {repoId: repo.id},
-                query: {ref: reference.id, after}
+                query: {
+                    ref: reference.id,
+                    after: (!!after) ? after : "",
+                    prefix: (!!prefix) ? prefix : "",
+                    delimiter: (!!delimiter) ? delimiter : "",
```

Comment:
no need to convert to boolean when we evaluate an expression already

```suggestion
                    after: after ? after : "",
                    prefix: prefix ? prefix : "",
                    delimiter: delimiter ? delimiter : "",
```